zebra/interface.c

```diff
@@ -1461,7 +1461,9 @@ static void interface_vrf_change(enum dplane_op_e op, ifindex_t ifindex,
 		 */
 		vrf_id_t exist_id = zebra_vrf_lookup_by_table(tableid, ns_id);
 
-		if (exist_id != VRF_DEFAULT) {
+		if (exist_id != VRF_DEFAULT ||
+		    (exist_id == VRF_DEFAULT &&
+		     strncmp(name, VRF_DEFAULT_NAME, strlen(VRF_DEFAULT_NAME)) != 0)) {
```

Comment:
Shouldn't we check an exact value (e.g. `strmatch()`)? What I meant is if we use a name "defaultX" it will match, right?